packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1285,8 +1310,10 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
       case MemoComponent:
       case SimpleMemoComponent:
       case Chunk: {
-        // Note: We currently never use MountMutation, but useLayout uses
-        // UnmountMutation.
+        // Note: We currently never use MountMutation, but useLayoutEffect uses UnmountMutation.
+        // This is to ensure ALL destroy fns are run before create fns,
+        // without requiring us to traverse the effects list an extra time during commit.
+        // This sequence prevents sibling destroy and create fns from interfering with each other.
```

Comment:
I added these comments for myself, since it wasn't clear to me *why* we were using these tags.